File: app/src/main/java/com/code/demo/demo/RetrofitUtilsActivity.java
package com.code.demo.demo;

import java.util.List;

import android.view.View;
import android.view.View.OnClickListener;
import android.widget.EditText;
import android.widget.TextView;

import com.alibaba.fastjson.JSON;
import com.code.demo.demo.model.LogisticsModel;
import com.code.demo.demo.presents.TestPresent;
import com.code.demo.demo.presents.TestPresent.ITest;
import com.code.fastframe.utils.ToastUtils;
import com.code.demo.R;
import com.code.fastframe.eventbus.GlobalEvent;
import com.code.fastframe.baseactivity.BaseTitleActivity;
import com.code.fastframe.widgets.SuperButton;


public class RetrofitUtilsActivity extends BaseTitleActivity implements ITest, OnClickListener {


    EditText mKey;

    SuperButton mRequest;

    TextView mList;

    TestPresent mTestPresent;

    @Override
    public String setTitleText() {
        return this.getClass().getSimpleName();
    }

    @Override
    public int setContentLayout() {
        return R.layout.activity_retrofitutils;
    }

    @Override
    public void initView() {
        mKey = findViewById(R.id.key);
        mRequest = findViewById(R.id.request);
        mList = findViewById(R.id.list);
        mRequest.setOnClickListener(this);
    }

    @Override
    public void initData() {
        super.initData();
        mTestPresent = new TestPresent(this, this);
    }


    @Override
    public void eventComming(GlobalEvent globalMsg) {

    }


    @Override
    public void showData(List<LogisticsModel> data) {
        mList.setText(JSON.toJSONString(data) + "");
    }

    @Override
    public void showError(String s) {
        ToastUtils.showToast(s);
    }

    @Override
    public void onClick(View v) {
        mTestPresent.getLogistics();
    }
}

File: app/src/main/java/com/code/demo/demo/RetrofitUtilsActivity.kt
package com.code.demo.demo

import android.view.View
import android.view.View.OnClickListener
import android.widget.EditText
import android.widget.TextView
import androidx.activity.viewModels
import androidx.lifecycle.Observer
import com.alibaba.fastjson.JSON
import com.code.demo.R.id
import com.code.demo.R.layout
import com.code.demo.viewmodels.RetrofitUtilsViewModel
import com.code.fastframe.baseactivity.BaseTitleActivity
import com.code.fastframe.eventbus.GlobalEvent
import com.code.fastframe.utils.L
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.android.synthetic.main.activity_retrofitutils.request

@AndroidEntryPoint
class RetrofitUtilsActivity : BaseTitleActivity() {
  var mKey: EditText? = null
  var mList: TextView? = null
  private val viewModel by viewModels<RetrofitUtilsViewModel>()
  override fun setTitleText(): String {
    return this.javaClass.simpleName
  }

  override fun setContentLayout(): Int {
    return layout.activity_retrofitutils
  }

  override fun initView() {
    mKey = findViewById(id.key)
    mList = findViewById(id.list)
    request.setOnClickListener {
      viewModel.requestData()
    }
    viewModel.liveData.observe(this, Observer {
      L.e("ui get")
      mList!!.text = JSON.toJSONString(it)
    })
  }

  override fun initData() {
    super.initData()
  }

  override fun eventComming(globalMsg: GlobalEvent) {}
}